cpp/src/io/parquet/reader_impl_chunking_utils.cuh

```diff
@@ -538,7 +544,7 @@ struct page_total_size {
   {
     // sum sizes for each input column at this row
     size_t sum = 0;
-    for (int idx = 0; idx < num_keys; idx++) {
+    for (auto idx = 0; std::cmp_less(idx, num_keys); idx++) {
```

Comment:
Not sure why the compiler didn't complain about this earlier but now fixed with `std::cmp`